nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -835,26 +836,35 @@ void Costmap2DROS::getCostsCallback(
   Costmap2D * costmap = layered_costmap_->getCostmap();
 
   for (const auto & pose : request->poses) {
-    bool in_bounds = costmap->worldToMap(pose.x, pose.y, mx, my);
+    geometry_msgs::msg::PoseStamped pose_transformed;
+    transformPoseToGlobalFrame(pose, pose_transformed);
+    bool in_bounds = costmap->worldToMap(
+      pose_transformed.pose.position.x,
+      pose_transformed.pose.position.y, mx, my);
 
     if (!in_bounds) {
-      response->costs.push_back(-1.0);
+      response->costs.push_back(costmap->getDefaultValue());
```

Comment:
Isn't unknown the better option here, if we're off the costmap? I don't think the default value applies where the map doesn't exist. Unknown seems rational to me since we really don't know anything